src/NzbDrone.Common/Http/HappyEyeballs/HttpHappyEyeballs.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using NLog;
+
+namespace NzbDrone.Common.Http.HappyEyeballs;
+
+public class HttpHappyEyeballs
+{
+    private const int ConnectionAttemptDelay = 250;
+
+    private readonly Logger _logger;
+
+    public HttpHappyEyeballs(Logger logger)
+    {
+        _logger = logger;
+    }
+
+    public async ValueTask<Stream> OnConnect(
+        SocketsHttpConnectionContext context,
+        CancellationToken cancellationToken)
+    {
+        var endPoint = context.DnsEndPoint;
+
+        var ipHostEntry = await Dns.GetHostEntryAsync(endPoint.Host, endPoint.AddressFamily, cancellationToken).ConfigureAwait(false);
+        var resolvedAddresses = ipHostEntry.AddressList;
+        if (resolvedAddresses.Length == 0)
+        {
+            throw new WebException(
+                $"The remote name {endPoint.Host} could not be resolved",
+                WebExceptionStatus.NameResolutionFailure);
+        }
+
+        var happyEyeballs = CreateHappyEyeballs(endPoint);
+        var socket = await happyEyeballs.Connect(resolvedAddresses, cancellationToken).ConfigureAwait(false);
+        _logger.Debug("Successfully connected {DnsEndPoint} to address: {RemoteEndPoint}", endPoint, socket.RemoteEndPoint);
```

Comment:
All these debug messages are _incredibly_ chatty, even at Trace I think they're too much.

Also while the tokens are replaced correctly, I think it makes more sense to stick with either numeric tokens or using ones that match the variables, the former being what the majority of the code base does right now.